File: Sources/JarvisNative/AppIdentity.swift
import Foundation

enum AppIdentity {
  static let appName = "Jarvis"
  static let voiceMessageHandlerName = "jarvisVoice"
  static let voiceBridgeObjectName = "jarvisVoiceBridge"

  static func applicationSupportRoot(fileManager: FileManager = .default) -> URL {
    let baseDirectory =
      fileManager.urls(for: .applicationSupportDirectory, in: .userDomainMask).first
      ?? URL(fileURLWithPath: NSHomeDirectory())
        .appending(path: "Library", directoryHint: .isDirectory)
        .appending(path: "Application Support", directoryHint: .isDirectory)
    return baseDirectory.appending(path: appName, directoryHint: .isDirectory)
  }

  static let resourceBundle: Bundle = {
    // Prefer SPM's generated resource bundle when available (swift build / debug),
    // fall back to the main app bundle (packaged .app).
    if let spmBundle = Bundle(url: Bundle.main.bundleURL
          .appending(path: "JarvisNative_JarvisNative.bundle")) {
      return spmBundle
    }
    return Bundle.main
  }()

  static func logsDirectory(fileManager: FileManager = .default) -> URL {
    let logsDirectory = applicationSupportRoot(fileManager: fileManager)
      .appending(path: "logs", directoryHint: .isDirectory)
    try? fileManager.createDirectory(at: logsDirectory, withIntermediateDirectories: true)
    return logsDirectory
  }
}

File: src/shared/appIdentity.ts
import fs from "node:fs";
import os from "node:os";
import path from "node:path";
import { APP_DISPLAY_NAME } from "./samanthaConfig";

export const APP_NAME = APP_DISPLAY_NAME;

export function resolveApplicationSupportRoot(): string {
  const baseDirectory = path.join(os.homedir(), "Library", "Application Support");
  return path.join(baseDirectory, APP_NAME);
}

export function resolveAppLogsDirectory(): string {
  const logsDirectory = path.join(resolveApplicationSupportRoot(), "logs");
  fs.mkdirSync(logsDirectory, { recursive: true });
  return logsDirectory;
}
